iceoryx_hoofs/include/iceoryx_hoofs/internal/concurrent/lockfree_queue/buffer.inl

```diff
@@ -14,6 +14,11 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
+#ifndef IOX_HOOFS_CONCURRENT_LOCKFREE_QUEUE_BUFFER_INL
+#define IOX_HOOFS_CONCURRENT_LOCKFREE_QUEUE_BUFFER_INL
+
+#include "iceoryx_hoofs/internal/concurrent/lockfree_queue/buffer.hpp"
```

Comment:
@elBoberido 
This is necessary to provide definitions locally in this file (but not for the way it is included). If not included, this causes clang-tidy warnings as it does not know about the context of inclusion.

This pseudo-cyclical inclusion (prevented by preprocessor) is weird design though.

Reply:
yes, it's required for clang-tidy but also other tools require it to cleanly parse the file